### ISSUE TITLE
Dropout causes extremely high CPU usage and slows down training

### ISSUE THREAD

**Issue by zhiwu111** [torchcvnn/torchcvnn#100]
When running the example `vit_mnist.py`, I noticed that the CPU usage becomes very high.

 After investigating, I found that the issue is caused by Dropout creating tensors on the **CPU** and then transferring them to the **GPU**, which introduces unnecessary overhead.

 After I made the following changes, the problem was resolved.

### `torchcvnn/nn/modules/dropout.py`

```python
class Dropout(nn.Module):
    def __init__(self, p: float = 0.5):
        super().__init__()
        self.p = p

    def forward(self, z: torch.Tensor) -> torch.Tensor:
        # Original code
        # mask = torch.nn.functional.dropout(
        #     torch.ones(z.shape), self.p, training=self.training
        # ).to(z.device)

        # Modified code
        mask = torch.nn.functional.dropout(
            torch.ones(z.shape, device=z.device), self.p, training=self.training
        )

        return mask * z


class Dropout2d(nn.Module):
    def __init__(self, p: float = 0.5):
        super().__init__()
        self.p = p

    def forward(self, z: torch.Tensor) -> torch.Tensor:
        # Original code
        # mask = torch.nn.functional.dropout2d(
        #     torch.ones(z.shape), self.p, training=self.training
        # ).to(z.device)

        # Modified code
        mask = torch.nn.functional.dropout2d(
            torch.ones(z.shape, device=z.device), self.p, training=self.training
        )
        return mask * z
```

------

### `torchcvnn/nn/functional.py`

```python
def dropout(
    z: Tensor, p: float = 0.5, training: bool = True, inplace: bool = False
) -> Tensor:
    if p < 0.0 or p > 1.0:
        raise ValueError(f"dropout probability has to be between 0 and 1, but got {p}")

    # Original code
    # mask = F.dropout(torch.ones(z.shape), p, training=training).to(z.device)

    # Modified code
    mask = F.dropout(torch.ones(z.shape, device=z.device), p, training=training)
    return mask * z
```

**Comment by jeremyfix** [comment]
Definitely a good point ,good catch, thank you . 

Would you like to propose a PR on this ?

**Comment by zhiwu111** [comment]
Sure, I‘ll submit a PR.
